File: articles/cascading-failure-and-contagion/go/cascade_summary.go
package main

import "fmt"

type Node struct {
	Sector    string
	Capacity  float64
	Load      float64
	Threshold float64
}

func main() {
	nodes := []Node{
		{"energy", 100, 62, 0.75},
		{"water", 85, 70, 0.70},
		{"telecom", 90, 58, 0.72},
		{"health", 95, 82, 0.78},
	}

	fmt.Println("Cascade capacity diagnostics")

	for _, node := range nodes {
		loadRatio := node.Load / node.Capacity
		status := "within threshold"
		if loadRatio >= node.Threshold {
			status = "failure risk"
		}
		fmt.Printf("%s load_ratio=%.3f threshold=%.3f status=%s\n", node.Sector, loadRatio, node.Threshold, status)
	}
}

File: articles/cascading-failure-and-contagion/rust/cascade_summary.rs
struct Node {
    sector: &'static str,
    capacity: f64,
    load: f64,
    threshold: f64,
}

fn main() {
    let nodes = vec![
        Node { sector: "energy", capacity: 100.0, load: 62.0, threshold: 0.75 },
        Node { sector: "water", capacity: 85.0, load: 70.0, threshold: 0.70 },
        Node { sector: "telecom", capacity: 90.0, load: 58.0, threshold: 0.72 },
        Node { sector: "health", capacity: 95.0, load: 82.0, threshold: 0.78 },
    ];

    println!("Cascade capacity diagnostics");

    for node in nodes {
        let load_ratio = node.load / node.capacity;
        let status = if load_ratio >= node.threshold { "failure risk" } else { "within threshold" };
        println!("{} load_ratio={:.3} threshold={:.3} status={}", node.sector, load_ratio, node.threshold, status);
    }
}
